tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/widgets/timegraph/widgets/TimeGraphControl.java

```diff
@@ -2333,8 +2345,21 @@ protected void drawMarker(IMarkerEvent marker, Rectangle bounds, ITimeDataProvid
             case SymbolType.CIRCLE:
                 SymbolHelper.drawCircle(gc, color, symbolSize, rect.x, y);
                 break;
-            default:
+            case SymbolType.DIAMOND:
                 SymbolHelper.drawDiamond(gc, color, symbolSize, rect.x, y);
+                break;
+            default:
+                char[] ca = symbol.toCharArray();
+                int first = ca[0];
+                /* check if the character is of length 1 or an emoji */
+                if (ca.length == 1 || (first >= EMOJI_LOW && first <= EMOJI_HIGH)) {
+                    Color oldColor = gc.getForeground();
+                    gc.setForeground(color);
+                    gc.drawText(symbol, rect.x, rect.y, true);
```

Comment:
The size of the emoji should be used to set the font:
TimeGraphRender.setFontForHeight(symbolSize * 2, gc);

The coordinates should be adjusted so that (rect.x, y) is the center of the emoji:
gc.drawText(symbol, rect.x - symbolSize, y - symbolSize, true);
This assumes that the emoji width is the same as its height.
It could be an ok assumption for a single character, otherwise we would need to calculate with gc.stringExtent (this is slow).

Reply:
> TimeGraphRender.setFontForHeight(symbolSize * 2, gc);

I used `(int) (rect.height * heightFactor)` directly instead of `symbolSize * 2` (which is `2 * (int) Math.ceil(rect.height * heightFactor / 2)`), otherwise we may get a slightly wrong value due to the conversion to `int` with `(int) Math.ceil(double)`.

> This assumes that the emoji width is the same as its height.
It could be an ok assumption for a single character, otherwise we would need to calculate with gc.stringExtent (this is slow).

Yeah, I think it's fine to try to horizontally center on the first emoji.

----

One thing I noticed is that an emoji seems to be drawn slightly bigger (specifically taller) than a normal character (same height factor of `1.0f`):

![image](https://github.com/eclipse-tracecompass/org.eclipse.tracecompass/assets/3717345/2b246574-e05c-4192-8346-b961cc113eea)

but I believe that it is pretty normal and expected: A📤. It can be adjusted using the height factor anyway.